app/src/components/layout/Navbar/Navbar.tsx

```diff
@@ -71,6 +73,20 @@ export const Navbar: React.FC = () => {
     };
   }, [isOpened]);
 
+  const handleNavItemSelection = (key: string) => {
```

Comment:
OK this might sound crazy but I actually think you could reduce the code here substantially. I don't actually think you need the `handleNavItemSelection` function, the `const [selectedRoute, setSelectedRoute] = useState('home');` or the `useEffect` below.

Since the `navItem` below is a `Link` from react-router-dom, as soon as you click on that it pushes the `path` to its `navigation` object, and we always have access to the pathname via the `useLocation` hook. This means that that hook is actually always controlling state, which we can just pull from like you do on like 80. So we could just do something like...

```
  const { pathname } = useLocation();
  const selectedRoute = pathname === '/' ? 'home' : pathname.slice(1);
```

And then get rid of the useState, useEffect and handleNavItemSelection.